common/stream.h

```diff
@@ -525,13 +557,13 @@ inline stream_out_t::stream_out_t()
 template<typename TType>
 inline void stream_out_t::push(const TType& value)
 {
-	if constexpr (std::is_trivial_v<TType> &&
+	if constexpr (std::is_trivially_copyable_v<TType> &&
 	              !(std::is_pointer_v<TType> ||
 	                std::is_reference_v<TType>))
```

Comment:
Could you also remove this part: ` !(std::is_pointer_v<TType> || std::is_reference_v<TType>)` due to it being redundant?

Template type deduction cannot result in a `TType` being a reference or a pointer type, since `value` is not of a a universal reference type